app/javascript/pages/Affiliates/Edit.tsx

```diff
@@ -0,0 +1,274 @@
+import { Icon } from "$app/components/Icons";
+import { NavigationButtonInertia } from "$app/components/NavigationButton";
+import { useLoggedInUser } from "$app/components/LoggedInUser";
+import { NumberInput } from "$app/components/NumberInput";
+import { showAlert } from "$app/components/server-components/Alert";
+import { PageHeader } from "$app/components/ui/PageHeader";
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "$app/components/ui/Table";
+
+type AffiliateProduct = {
+  id: number;
+  name: string;
+  enabled: boolean;
+  fee_percent: number | null;
+  destination_url: string | null;
+  referral_url: string;
+};
+
+type AffiliateData = {
+  id: string;
+  email: string;
+  destination_url: string | null;
+  affiliate_user_name: string;
+  fee_percent: number;
+  products: AffiliateProduct[];
+};
+
+type Props = {
+  affiliate: AffiliateData;
+  affiliates_disabled_reason: string | null;
+};
+
+export default function AffiliatesEdit() {
+  const props = usePage<{ props: Props }>().props as unknown as Props;
+  const loggedInUser = useLoggedInUser();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [products, setProducts] = React.useState<AffiliateProduct[]>(props.affiliate.products);
+  const [feePercent, setFeePercent] = React.useState<number | null>(props.affiliate.fee_percent);
+  const [destinationUrl, setDestinationUrl] = React.useState<string | null>(props.affiliate.destination_url);
+  const [errors, setErrors] = React.useState<Map<string, string>>(new Map());
+
+  const applyToAllProducts = products.every((p) => p.enabled && p.fee_percent === feePercent);
+
+  const uid = React.useId();
+
+  const toggleAllProducts = (checked: boolean) => {
+    if (checked) {
+      setProducts(products.map((p) => ({ ...p, enabled: true, fee_percent: feePercent })));
+    } else {
+      setProducts(products.map((p) => ({ ...p, enabled: false })));
+    }
+  };
+
+  const handleSubmit = asyncVoid(async () => {
+    const newErrors = new Map<string, string>();
+
+    if (applyToAllProducts && (!feePercent || feePercent < 1 || feePercent > 90)) {
+      newErrors.set("feePercent", "Commission must be between 1% and 90%");
+    }
+
+    if (!applyToAllProducts && products.every((p) => !p.enabled)) {
+      newErrors.set("products", "Please enable at least one product");
+    }
+
+    if (!applyToAllProducts && products.some((p) => p.enabled && (!p.fee_percent || p.fee_percent < 1 || p.fee_percent > 90))) {
+      newErrors.set("products", "All enabled products must have commission between 1% and 90%");
+    }
+
+    if (destinationUrl && destinationUrl !== "" && !isUrlValid(destinationUrl)) {
+      newErrors.set("destinationUrl", "Please enter a valid URL");
+    }
+
+    if (newErrors.size > 0) {
+      setErrors(newErrors);
+      const [, firstError] = Array.from(newErrors.entries())[0] || [];
+      if (firstError) showAlert(firstError, "error");
+      return;
+    }
```

Comment:
Not blocking for this PR, but this should now ideally be handled server-side since [Inertia + Rails can do it](https://inertia-rails.dev/guide/validation#how-it-works)